index.js

```diff
@@ -209,6 +258,9 @@ module.exports = {
             var oclick = nemo.wd.WebElement.prototype.click;
             nemo.wd.WebElement.prototype.click = function () {
                 var filename = 'ScreenShot_onClick-' + process.pid + '-' + new Date().getTime();
+                if (autoCaptureOptions.indexOf('source') !== -1) {
```

Comment:
this call should be added to the existing promise chain to make sure its errors are handled in the existing catchers.

```js
let sourcePromise = (autoCaptureOptions.indexOf('source') !== -1) ? nemo.screenshot.source(filename) : Promise.resolve(true);
return sourcePromise
    .then(nemo.screenshot.snap(filename))
    .then(oclick.apply(this, arguments));
```

I think that should work.